File: contains-duplicate-121/contains_duplicate.java
class Solution {
    public boolean containsDuplicate(int[] nums) {
        HashMap<Integer,Integer> seen = new HashMap<>();

        for(int i=0; i < nums.length; i++) {
            if(seen.containsKey(nums[i])) {
                return true;
            }
            seen.put(nums[i], 1);
        }

        return false;
    }
}

File: contains-duplicate-121/contains_duplicate.rs
use std::collections::HashSet;

impl Solution {
    pub fn contains_duplicate(nums: Vec<i32>) -> bool {
        let mut seen = HashSet::new();
        for num in nums.iter() {
        if !seen.insert(*num) {
            return true;
        }
    }
    false
}
}